mindsdb/integrations/handlers/strava_handler/strava_handler.py

```diff
@@ -0,0 +1,102 @@
+from mindsdb.integrations.handlers.strava_handler.strava_tables import StravaAllClubsTable
+from mindsdb.integrations.handlers.strava_handler.strava_tables import StravaClubActivitesTable
+from mindsdb.integrations.libs.api_handler import APIHandler
+from mindsdb.integrations.libs.response import (
+    HandlerStatusResponse as StatusResponse,
+)
+from mindsdb.utilities.log import get_log
+from mindsdb_sql import parse_sql
+
+import requests
+import pandas as pd
+import json
+
+logger = get_log("integrations.stravahandler")
+
+class StravaHandler(APIHandler):
+    """Strava handler implementation"""
+
+    def __init__(self, name=None, **kwargs):
+        """Initialize the Confluence handler.
+        Parameters
+        ----------
+        name : str
+            name of a handler instance
+        """
+        super().__init__(name)
+
+        connection_data = kwargs.get("connection_data", {})
+
+        self.parser = parse_sql
+        self.dialect = 'strava'
+        self.connection_data = connection_data
+        self.kwargs = kwargs
+        self.connection = None
+        self.is_connected = False
+
+        strava_all_clubs_data = StravaAllClubsTable(self)
+        self._register_table("all_clubs", strava_all_clubs_data)
+        
+        strava_club_activites_data = StravaClubActivitesTable(self)
+        self._register_table("club_activities", strava_club_activites_data)
+
+    def connect(self) -> StatusResponse:
+        """Set up the connection required by the handler.
+        Returns
+        -------
+        StatusResponse
+            connection object
+        """
+        if self.is_connected is True:
+            return self.connection
+
+        strava_url = f"https://www.strava.com/api/v3/athlete/clubs"
+        headers = {
+            "Authorization": f"Bearer {self.connection_data['strava_api_token']}",
+        }
+        response = requests.request("GET",strava_url,headers=headers)
```

Comment:
please add a timeout to any call of 'requests' (5 or 10 seconds?)

Reply:
Dropped requests & implemented the handler with strava python API lib